[weed/s3] `chunk-signature` and `x-amz` headers appear within stored file content

See https://github.com/andrewazores/quarkus-3.20-s3-chunk-encode for a reproducer. This is a stripped-down Java test application built on Quarkus, using the AWS Java SDK S3Client to communicate with the object storage instance. The test is simple:

1. `POST /test` a simple `hello world` text file to the application. This in turn issues a `PutObjectRequest` to the storage container. This operation appears to succeed.
2. `GET /` on the application, which issues a `ListObjectsRequest` to the storage container. This returns the expected result of a JSON array with the uploaded file key `["test"]`.
3. `GET /test` on the application, which issues a `GetObjectRequest` to the storage container. This returns the stored file content back. This should obviously be identical to the `hello world` text file which was uploaded.

The test matrix so far:

| Quarkus version | AWS SDK version | Object Storage | Result |
| - | - | - | - |
| 3.15 | 2.27.20 | SeaweedFS 3.88 | Pass |
| 3.19 | 2.29.14 | SeaweedFS 3.88 | Pass |
| 3.20 | 2.30.36 | SeaweedFS 3.63 | **Fail** |
| 3.20 | 2.30.36 | SeaweedFS 3.80 | **Fail** |
| 3.20 | 2.30.36 | SeaweedFS 3.84 | **Fail** |
| 3.20 | 2.30.36 | SeaweedFS 3.88 | **Fail** |
| 3.20 | 2.30.36 | SeaweedFS 3.89 | **Fail** |
| 3.15 | 2.27.20 | Localstack 3.8.1 | Pass |
| 3.20 | 2.30.36 | Localstack 3.8.1 | Pass |
| 3.15 | 2.27.20 | Localstack 4.1.1 | Pass |
| 3.20 | 2.30.36 | Localstack 4.1.1 | Pass |
| 3.20 | 2.30.36 | Localstack 4.5.0 | Pass |

The failing result looks like this:

```
[ERROR] Failures: 
[ERROR]   StorageResourceWithCryostatStorageTest.testGet:74 
Expected: "hello world\n"
     but: was "c;chunk-signature=347f6c62acd95b7c6ae18648776024a9e8cd6151184a5e777ea8e1d9b4e45b3c\r\nhello world\n\r\n0;chunk-signature=1a99b7790b8db0f4bfc048c8802056c3179d561e40c073167e79db5f1a6af4b2\r\nx-amz-checksum-crc32:rwg7LQ==\r\nx-amz-trailer-signature:92c52be17274657984b73b3046f1d36c12a351a9ef7ae668226f17f9d32f42aa\r\n\r\n"
```

The file content retrieved from Seaweed is *not* the identical file content which was originally uploaded, but instead has these `chunk-signature`, `x-amz-checksum-crc32`, and `x-amz-trailer-signature` headers(?) included.

This seems related to or maybe a duplicate of https://github.com/seaweedfs/seaweedfs/issues/6583 , but in this case we are not dealing with any HTTPS or any proxies.

(the actual SeaweedFS container in the test is from https://github.com/cryostatio/cryostat-storage, but as you can see from [its Dockerfile](https://github.com/cryostatio/cryostat-storage/blob/main/Dockerfile) and [entrypoint script](https://github.com/cryostatio/cryostat-storage/blob/main/cryostat-entrypoint.bash) this is a very vanilla build of SeaweedFS)

If I disable chunked encoding on the S3 client instance (`quarkus.s3.chunked-encoding=false`) then this **passes**. I don't know SeaweedFS or the S3 client well enough to know if there has been some change to the chunked encoding handling on either side or both sides.

--- a/weed/s3api/chunked_reader_v4.go
+++ b/weed/s3api/chunked_reader_v4.go
@@ -520,15 +520,14 @@ func parseChunkChecksum(b *bufio.Reader) (ChecksumAlgorithm, []byte, error) {
 			return ChecksumAlgorithmNone, nil, err
 		}
 
-		line := trimTrailingWhitespace(bytesRead)
-		if len(line) == 0 {
+		if len(bytesRead) == 0 {
 			break
 		}
 
-		parts := bytes.SplitN(line, []byte(":"), 2)
+		parts := bytes.SplitN(bytesRead, []byte(":"), 2)
 		if len(parts) == 2 {
-			key := string(parts[0])
-			value := trimTrailingWhitespace(parts[1])
+			key := string(bytes.TrimSpace(parts[0]))
+			value := bytes.TrimSpace(parts[1])
 			if alg, err := extractChecksumAlgorithm(key); err == nil {
 				if checksumAlgorithm != ChecksumAlgorithmNone {
 					glog.V(3).Infof("multiple checksum headers found in trailer, using last: %s", key)

